src/Sarif/UriHelper.cs

```diff
@@ -69,6 +71,16 @@ public static string MakeValidUri(string path)
                     !path.StartsWith(@"\", StringComparison.Ordinal))
                 {
                     validUri = validUri.Substring(1);
+
+                    // When the UriBuilder constructs an absolute URI, it strips any
+                    // leading "." and ".." segments ("dot-segments", as RFC 3986 calls
+                    // them). Glue them back on so we don't lose the relative path
+                    // information.
+                    string leadingDotSegments = GetLeadingDotSegments(path);
+                    if (!string.IsNullOrEmpty(leadingDotSegments))
+                    {
+                        validUri = leadingDotSegments + validUri;
+                    }
```

Comment:
Sorry, the comment above was intended to explain this code path. Without this code, the "URI" `..\..\..\foo\bar\baz.c` serializes as `foo/bar/baz.c`, losing the "dot-segments" (the relative path information). With this code, that URI serializes as `../../../foo/bar/baz.c` (and the new unit test verifies that). I'll add that example to the comment.

We only found that because SDV emits paths of that form. If Roslyn never does, there's no bug. Is it possible for Roslyn to hit this?